packages/components/src/InputTime/InputTime.rebuilt.tsx

```diff
@@ -1,101 +1,114 @@
-import type { InputTimeProps, InputTimeRebuiltProps } from "./InputTime.types";
-import { dateToTimeString, timeStringToDate } from "./utils/input-time-utils";
+import { useInputTimeActions } from "./hooks/useInputTimeActions";
+import type { InputTimeRebuiltProps } from "./InputTime.types";
+import { dateToTimeString } from "./utils/input-time-utils";
 import { FormFieldWrapper, useFormFieldWrapperStyles } from "../FormField";
+import { mergeRefs } from "../utils/mergeRefs";
+
+export const InputTimeRebuilt = forwardRef<
+  HTMLInputElement,
+  InputTimeRebuiltProps
+>(function InputTimeRebuilt(
+  {
+    value,
+    onChange,
+    readOnly,
+    autoComplete,
+    // Deprecated props
+    inputRef: deprecatedInputRef,
+    ...props
+  },
+  forwardedRef,
+) {
+  const internalRef = useRef<HTMLInputElement>(null);
+  const mergedRef = mergeRefs<HTMLInputElement>([
+    internalRef,
+    deprecatedInputRef as React.RefObject<HTMLInputElement>,
+    forwardedRef,
+  ]);
+  const id = props.id || useId();
+  const wrapperRef = React.useRef<HTMLDivElement>(null);
+  const { inputStyle } = useFormFieldWrapperStyles(props);
+
+  const {
+    handleChangeEvent,
+    handleChange,
+    handleBlur,
+    handleClear,
+    handleFocus,
+    handleKeyDown,
+  } = useInputTimeActions({
+    onChange,
+    value,
+    readOnly,
+    disabled: props.disabled,
+    inputRef: internalRef,
+    onFocus: props.onFocus,
+    onBlur: props.onBlur,
+    onKeyDown: props.onKeyDown,
+  });
 
-export function InputTimeRebuilt({
-  value,
-  onChange,
-  ...params
-}: InputTimeRebuiltProps) {
-  const ref =
-    (params.inputRef as React.RefObject<HTMLInputElement>) ??
-    useRef<HTMLInputElement>(null);
   const { setTypedTime } = useTimePredict({
     value,
     handleChange,
   });
 
-  const { inputStyle } = useFormFieldWrapperStyles(params);
+  // Kept outside the useInputTimeActions hook to avoid circular dependency via setTypedTime and handleChange
+  function handleKeyUp(event: React.KeyboardEvent<HTMLInputElement>) {
+    props.onKeyUp?.(event);
+    if (props.disabled || readOnly) return;
 
-  const wrapperRef = React.useRef<HTMLDivElement>(null);
-  const id = getId(params);
+    !isNaN(parseInt(event.key, 10)) && setTypedTime(prev => prev + event.key);
+  }
+
+  const isInvalid = Boolean(props.error || props.invalid);
 
   return (
     <FormFieldWrapper
-      disabled={params.disabled}
-      size={params.size}
-      align={params.align}
-      inline={params.inline}
-      name={params.name}
-      error={params.error || ""}
+      disabled={props.disabled}
+      size={props.size}
+      align={props.align}
+      inline={props.inline}
+      name={props.name}
+      error={props.error || ""}
       identifier={id}
       descriptionIdentifier={`descriptionUUID--${id}`}
-      invalid={Boolean(params.invalid)}
-      description={params.description}
-      clearable={params.clearable ?? "never"}
+      invalid={Boolean(props.invalid)}
```

Comment:
```suggestion
      invalid={props.invalid}
```

You already convert it to Boolean on [line 65](https://github.com/GetJobber/atlantis/pull/2805/files#diff-5c5a812d081c6e3e0e0f58a5637b543eae60420bf45219acf03085c14b01c1e7R65).

Reply:
Although, it's slightly different. You define `isInvalid`:
```tsx
const isInvalid = Boolean(props.error || props.invalid);`
```

So, perhaps, what you have is correct, unless it should be `invalid={isInvalid}`